modules/meeting/app/views/meetings/_form.html.erb

```diff
@@ -35,6 +35,23 @@ See COPYRIGHT and LICENSE files for more details.
     <%= f.text_field :title, :required => true, :size => 60, container_class: '-wide' %>
   </div>
 
+  <% if global_create_context? %>
+    <div class="form--field -required">
+      <%= f.select :project_id,
+                   options_for_project_selection,
+                   {prompt: t(:project_selection_placeholder),
+                    container_class: '-wide'},
+                   class: project_select_initial_class_list,
+                   required: true,
+                   data: {
+                     'application-target': 'dynamic',
+                     controller: 'select-field-with-prompt',
+                     action: 'change->select-field-with-prompt#togglePromptStyling ' \
+                              'change->refresh-on-form-changes#triggerTurboReload'
+                   } %>
```

Comment:
There is an angular powered autocompleter that sports all the niceties of a typeahead and also supports hierarchies:

<img width="589" alt="image" src="https://github.com/opf/openproject/assets/617519/7a644c4b-7740-4667-ad20-680b71d25b17">

It is e.g. used on the [page for bulk moving work packages](https://github.com/opf/openproject/blob/dev/app/views/work_packages/moves/new.html.erb#L65-L77). It should also be used here.

Doing this will require sending the correct filters for the all projects, in which the user is allowed to create meetings. Please see the [page for bulk moving work packages (the `apiFilters` property)](https://github.com/opf/openproject/blob/dev/app/views/work_packages/moves/new.html.erb#L67) as an example again. In order for that to work, the [`create_meetings` permission](https://github.com/opf/openproject/blob/dev/modules/meeting/lib/open_project/meeting/engine.rb#L43) needs to have `contract_actions` added, see [project create](https://github.com/opf/openproject/blob/dev/config/initializers/permissions.rb#L36) for an example.